tools/session-ui/webconsoleServer.rb

```diff
@@ -0,0 +1,130 @@
+# This Module will act as an intermediate between metasploit console and meterpreter WebConsole.
+# it will initiate WebConsole server for a specific session. Glue code present in this Class will
+# fetch lists of post module from msfconsole in json format and will be converted in a format that can
+# be readable by the browser.
+
+require 'sinatra/base'
+require 'json'
+
+
+module Server
+  class WebConsoleServer < Sinatra::Base
+
+    configure :development do
+      #set :root, File.dirname(__FILE__)
+      set :json_content_type, :js
+      set :public_folder, File.dirname(__FILE__)+'/public'
+      set :bind,'127.0.0.1'
+      set :server, %w[thin mongrel webrick]
+      set :content_type,'json'
+
+    end
+
+
+    get '/' do
+      # receives an input from
+      File.open(File.join(File.dirname(__FILE__)+'/public','public.html'))
+    end
+
+
+    get "/sysinfo" do
+      system_info=File.read(File.join(File.dirname(__FILE__),'sysinfo.json'))
+      return system_info
+    end
+
+    post "/modal" do
+      content_type :json
+
+    end
+
+    get "/post" do
```

Comment:
Please make the endpoints more descriptive. `post-modules`, `extensions` are reasonable length and more clear as to what is expected to be returned.